src/main/java/br/com/deveficiente/bolaoapi/services/user/User.java

```diff
@@ -0,0 +1,61 @@
+package br.com.deveficiente.bolaoapi.services.user;
+
+import lombok.Getter;
+import lombok.ToString;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+@ToString
+@Entity(name = "poll_user")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Getter
+    private Long id;
+
+    @NotBlank
+    @Email(regexp = ".+@.+\\..+")
+    @Getter
+    private String login;
+
+    @NotBlank
+    @Getter
+    private String password;
+
+    @NotNull
+    @Getter
+    private LocalDateTime timestamp;
+
+    private static BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+
+    public User() {
+    }
+
+    public User(@NotBlank String login, @NotBlank String rawPassword) {
```

Comment:
O nome do parâmetro já ajuda! É isso aí :). Algumas outras opções que poderia rolar:

```
/**
 @param login precisa ser o login no formato de email
 @param senha precisa ser uma senha em plain text
**/
public User(@NotBlank @Email String login, @NotBlank String rawPassword)
```

Outro esquema usando uma classe para deixar mais claro. 

```
 public User(@NotBlank String login, RawPassword rawPassword){
   this.login = login;
   this.password = rawPassword.encode();
 }

 ```

Reply:
Gostei do 2 esquema, vou pelo primeiro para forçar na minha cabeça a não criação de uma nova classe.